### ISSUE TITLE
Can't use since last update

### ISSUE THREAD

**Issue by natcl** [skonagaya/HTTP-Push#8]
Since the last update it doesn't seem to work anymore.  Watch keeps telling me to resend from my phone which doesn't seem to work.

Thanks !

**Comment by skonagaya** [comment]
What do you see on the watch when you click on "Save and Exit" on the phone? Also which pebble are you using?

**Comment by natcl** [comment]
I am using pebble round.
When I go in the App on my watch it says
"NEW VERSION DETECTED Resend request from your phone"

And then if I click Save and exit on the phone it says loading on the watch and goes back to the homescreen. If I trigger the app again it still says "NEW VERSION DETECTED"

**Comment by skonagaya** [comment]
Can you send me your backup string via email so I can reproduce your issue on my watch? 

Copy the **Copy Data** string in the **Backup** tab in the phone, and email via **Report** bug tab.

**Comment by natcl** [comment]
Is there a trick to select all the text ? I'm having a hard time selecting the whole text as it won't scroll up or down

**Comment by natcl** [comment]
I could'nt send it with the report tab as I don't have an email client configured on my phone (I use inbox)
here's the data:

```
[{"type":"folder","name":"TV","list":[{"type":"request","name":"Power","endpoint":"http://192.168.1.78:1880/tv","json":"{\"command\": \"power\"}","method":"POST","toDelete":false}],"toDelete":false},{"type":"folder","name":"Musique","list":[{"type":"request","name":"Play/Pause","endpoint":"http://192.168.1.78:1880/itunes","json":"{\"command\": \"playpause\"}","method":"POST","toDelete":false},{"type":"request","name":"Volume up","endpoint":"http://192.168.1.78:1880/itunes","json":"{\"command\": \"volumeup\"}","method":"POST","toDelete":false},{"type":"request","name":"Volume down","endpoint":"http://192.168.1.78:1880/itunes","json":"{\"command\": \"volumedown\"}","method":"POST","toDelete":false},{"type":"folder","name":"Radios","list":[{"type":"request","name":"Première Chaîne","endpoint":"http://192.168.1.78:1880/radio","json":"{\"command\": \"Première Chaîne\"}","method":"POST","toDelete":false},{"type":"request","name":"Espace Musique","endpoint":"http://192.168.1.78:1880/radio","json":"{\"command\": \"Espace Musique\"}","method":"POST","toDelete":false},{"type":"request","name":"FIP","endpoint":"http://192.168.1.78:1880/radio","json":"{\"command\": \"FIP\"}","method":"POST","toDelete":false}],"toDelete":false}],"toDelete":false},{"type":"request","name":"Bixi","endpoint":"http://192.168.1.78:1880/bixi","json":"[1]","method":"POST","toDelete":false},{"type":"request","name":"Powerhjj","endpoint":"http://192.168.1.78:1880/tv","json":"{\"command\": \"power\"}","method":"POST","toDelete":false}]
```

**Comment by skonagaya** [comment]
Thanks. I was able to reproduce the error. I'll let you know when this is resolved.

**Comment by skonagaya** [comment]
There seems to be a problem with the size of the request. I removed 2 of your requests and it seems to work..

Failing here even though it was free'd and set to NULL prior...
`s_buffer = (char*)malloc((length+1) * sizeof(char));`

Still checking..

**Comment by skonagaya** [comment]
Root cause seems to be the use of diacritics (accents). 

For now, it should work if you replace "Premi**è**re Cha**î**ne" with "Premi**e**re Cha**i**ne".

**Comment by natcl** [comment]
Thanks ! Worked great !

**Comment by skonagaya** [comment]
This is interesting actually.. Diacritics/accented characters return the length of 2 instead of 1. So the character è returns the length for "e" and "`". A combination of characters. 

http://stackoverflow.com/a/18578226

Anyways I'll create another release to cover those characters. Bear with me.

**Comment by natcl** [comment]
Thanks !
Funny thing is that I seem able to leave them in the actual json data of the command I send but not in the name of the command.

**Comment by skonagaya** [comment]
That sounds about right. The app only needs the names of the folders/requests. So the json data stays on the phone. Think of the watch being a remote for your phone.

Anyways this has been addressed in 3b7e32db7fa1a4bab64e0bcabaec58afd0dcc926. 

The update:
`unescape(encodeURIComponent(listToString)).length`
